gloo/transport/tcp/unbound_buffer.cc

```diff
@@ -72,11 +84,12 @@ void UnboundBuffer::waitRecv(int* rank, std::chrono::milliseconds timeout) {
                   "ms for recv operation to complete"));
     }
   }
-
+  auto recvCompleted = recvCompletions_ > 0;
```

Comment:
If it was aborted, it should return here and not try and decrement `recvCompletions_`. This counter is a tally that counts the number of recv's that completed without an accompanying `waitRecv` call. If it is non-zero when `waitRecv` is called, `waitRecv` can return immediately without blocking (as evidenced by the condition `if (recvCompletions_ == 0)`).